packages/docusaurus/src/server/__tests__/configValidation.test.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as Joi from '@hapi/joi';
+import {ConfigSchema, DEFAULT_CONFIG} from '../configValidation';
+
+describe('validateConfig', () => {
+  test('normalize config', () => {
+    const value = Joi.attempt(
+      {
+        baseUrl: '/',
+        favicon: 'some.ico',
+        title: 'my site',
+        url: 'https://mysite.com',
+      },
+      ConfigSchema,
+    );
+    expect(value).toEqual({
+      ...DEFAULT_CONFIG,
+      baseUrl: '/',
+      favicon: 'some.ico',
+      title: 'my site',
+      url: 'https://mysite.com',
+    });
+  });
+
+  test('throw error for unknown field', () => {
+    expect(() => {
+      Joi.attempt(
+        {
+          baseUrl: '/',
+          favicon: 'some.ico',
+          title: 'my site',
+          url: 'https://mysite.com',
+          invalid: true,
+        },
+        ConfigSchema,
+      );
+    }).toThrowErrorMatchingSnapshot();
+  });
+
+  test('throw error for baseUrl without trailing `/`', () => {
+    expect(() => {
+      Joi.attempt(
+        {
+          baseUrl: 'noslash',
+          favicon: 'some.ico',
+          title: 'my site',
+          url: 'https://mysite.com',
+        },
+        ConfigSchema,
+      );
+    }).toThrowErrorMatchingSnapshot();
+  });
+
+  test('throw error if plugins is not array', () => {
+    expect(() => {
+      Joi.attempt(
+        {
+          baseUrl: '/',
+          favicon: 'some.ico',
+          title: 'my site',
+          url: 'https://mysite.com',
+
+          plugins: {},
+        },
+        ConfigSchema,
+      );
+    }).toThrowErrorMatchingSnapshot();
+  });
+
+  test('throw error if themes is not array', () => {
+    expect(() => {
+      Joi.attempt(
+        {
+          baseUrl: '/',
+          favicon: 'some.ico',
+          title: 'my site',
+          url: 'https://mysite.com',
+
+          themes: {},
+        },
+        ConfigSchema,
+      );
+    }).toThrowErrorMatchingSnapshot();
+  });
+
+  test('throw error if presets is not array', () => {
+    expect(() => {
+      Joi.attempt(
+        {
+          baseUrl: '/',
+          favicon: 'some.ico',
+          title: 'my site',
+          url: 'https://mysite.com',
+
+          presets: {},
+        },
+        ConfigSchema,
+      );
+    }).toThrowErrorMatchingSnapshot();
+  });
+
+  test("throw error if scripts doesn't have src", () => {
+    expect(() => {
+      Joi.attempt(
+        {
+          baseUrl: '/',
+          favicon: 'some.ico',
+          title: 'my site',
+          url: 'https://mysite.com',
```

Comment:
What about extracting this in a "baseConfig" variable?

Maybe it could be more convenient to have such method to make tests less "verbose"


```
const testConfig = (config) => Joi.attempt({...baseConfig,...config})
```

Reply:
Good idea , I will fix it.